docs/wiki/concepts/CSP-Node.md

```diff
@@ -0,0 +1,276 @@
+A basket is essentially a collection of timeseries which can be passed in as a single argument.
+Baskets can either be list baskets or dict baskets.
+Individual timeseries in a basket can tick independently, and they can be looked at and reacted to individually or as a collection.
+
+For example:
+
+```python
+@csp.node                                      # 1
+def demo_basket_node(                          # 2
+    list_basket: [ts[int]],                    # 3
+    dict_basket: {str: ts[int]}                # 4
+) -> ts[float]:                                # 5
+                                               # 6
+    if csp.ticked(list_basket):                # 7
+        return sum(list_basket.validvalues())  # 8
+                                               # 9
+    if csp.ticked(list_basket[3]):             # 10
+        return list_basket[3]                  # 11
+                                               # 12
+    if csp.ticked(dict_basket):                # 13
+        # can iterate over ticked key,items    # 14
+        # for k,v in dict_basket.tickeditems():# 15
+        #     ...                              # 16
+        return sum(dict_basket.tickedvalues()) # 17
+```
+
+3\) Note the syntax of basket inputs.
+list baskets are noted as `[ts[type]]` (a list of time series) and dict baskets are `{key_type: ts[ts_type]}` (a dictionary of timeseries keyed by type `key_type`).
+
+7\) Just like single timeseries, we can react to a basket if it ticked.
+The convention is the same as passing multiple inputs to `csp.ticked`, `csp.ticked` is true if **any** basket input ticked.
+`csp.valid` is true is **all** basket inputs are valid.
+
+8\) baskets have various iterators to access their inputs:
+
+- **`tickedvalues`**: iterator of values of all ticked inputs
+- **`tickedkeys`**: iterator of keys of all ticked inputs (keys are list index for list baskets)
+- **`tickeditems`**: iterator of (key,value) tuples of ticked inputs
+- **`validvalues`**: iterator of values of all valid inputs
+- **`validkeys`**: iterator of keys of all valid inputs
+- **`validitems`**: iterator of (key,value) tuples of valid inputs
+- **`keys`**: list of keys on the basket (**dictionary baskets only** )
+
+10-11) This demonstrates the ability to access an individual element of a
+basket and react to it as well as access its current value
+
+## **Node Outputs**
+
+Nodes can return any number of outputs (including no outputs, in which case it is considered an "output" or sink node,
+see [Graph Pruning](https://github.com/Point72/csp/wiki/0.-Introduction#graph-pruning)).
+Nodes with single outputs can return the output as an unnamed output.
+Nodes returning multiple outputs must have them be named.
+When a node is called at graph building time, if its is a single unnamed node the return variable is an edge representing the output which can be passed into other nodes.
```

Comment:
Is there any place in the docs that mentions an output timeseries cannot be ticked more than once in a given node invocation?  We should mention it somewhere if not, and this seems like a good spot

Reply:
Added by Melissa, resolving